src/common/Threading/LockedQueue.h

```diff
@@ -18,74 +18,67 @@
 #ifndef LOCKEDQUEUE_H
 #define LOCKEDQUEUE_H
 
+#include <atomic>
 #include <deque>
+#include <memory>
 #include <mutex>
 
 template <class T, typename StorageType = std::deque<T>>
 class LockedQueue
 {
-    //! Lock access to the queue.
-    std::mutex _lock;
+    mutable std::mutex _lock;
 
-    //! Storage backing the queue.
-    StorageType _queue;
+    std::atomic<bool> _canceled{false};
 
-    //! Cancellation flag.
-    volatile bool _canceled{false};
+    StorageType _queue;
 
 public:
 
-    //! Create a LockedQueue.
+    // Create a LockedQueue.
```

Comment:
I would like the comments to use three slashes `///` this way Doxygen will pick it up